### ISSUE TITLE
Very high memory consumption + GC cpu time (selector.select not blocking)

### ISSUE THREAD

**Issue by darkwitch** [Shevchik/UdpServerSocketChannel#1]
I'm using your UdpServerSocketChannel in my project, but I notice very high cpu load after starting the application on both Windows and Linux (Ubuntu 16.04).

Minimal code required to reproduce: your ExampleUdpServer

When you lunch it this is what happens (both when app is just lunched waiting for connections and after few connections were made):

![image](https://cloud.githubusercontent.com/assets/10908606/23849796/33879f46-07dd-11e7-9944-6a4572776465.png)

Which is caused by the nioEventLoopGroup thread pool:

![image](https://cloud.githubusercontent.com/assets/10908606/23850120/6d62c280-07de-11e7-8278-b544c7720e9c.png)

In general the application is producing lots of ArrayList.Iterator :) and GC is taking lot's of cpu time to clean it up.

![image](https://cloud.githubusercontent.com/assets/10908606/23850144/85d65264-07de-11e7-9453-1cc77c8d7546.png)

I've traced it to selector.select(timeoutMillis) call of netty 4.0.30.Final (NioEventLoop:622)
![image](https://cloud.githubusercontent.com/assets/10908606/23849923/9c1fd0d2-07dd-11e7-86f3-0304d562972f.png)

In my opinion this should block until new operation is ready for the DatagramChannel, but in case of netty and your plugin this doesn't happen.

Here's a non-netty sample where this behaves correctly - might be usefull.
http://thushw.blogspot.com/2011/06/asynchronous-udp-server-using-java-nio.html

I've modified your code to make it compatible with netty 4.1.1.Final - but the problem persists.
This doesn't happen when using netty with TCP.

**Comment by darkwitch** [comment]
Same happens on current netty 4.0.45.Final

**Comment by Shevchik** [comment]
Does it happen just after start (while idling), or when you have an active connection?

**Comment by darkwitch** [comment]
Actually both :)

Just after start while idling. 
The same for active connections.

**Comment by Shevchik** [comment]
I made some changes that may fix this issue, try it.

**Comment by darkwitch** [comment]
![image](https://cloud.githubusercontent.com/assets/10908606/23856983/d388fa1c-07fb-11e7-8b81-752488deeedc.png)

WOOT!

https://www.youtube.com/watch?v=AXwGVXD7qEQ

It's working... It's working ;)

Checked on windows & linux. 

Strangely - the culprit seems to be https://github.com/Shevchik/UdpServerSocketChannel/commit/3ab852061fce5931974ec10b24eb2fcc956bbe92 , which is kinda weird.

Too bad there're no automated testing, I've fired some of mine and everything seems to work ok, but that's not nearly enough coverage
